app/uid/models.py

```diff
@@ -0,0 +1,130 @@
+
+    #def get_lcvterms(self):
+        #return self.lcvterms.all()
+
+    #def get_upstream(self)
+        #providers = []
+        #current_node = self
+        #while current_node:
+            #provider_nodes = currrent_node.get_providers()
+            #if provider_nodes:
+            #   provider_node = parent_nodes[0]
+            # providers.append(provider_node)
+            # current_node = provider_node
+            # else:
+            #       current_node = None
+        #return providers
+    
+    #def get_downstream(self):
+        #lcv_terms = []
+        #nodes_to_visit = [self]
+        #while nodes_to_visit:
+            #current_node = nodes_to_visit.pop()
+            #lcv_terms_nodes = current_node.get_lcv_terms()
+            #lcv_terms.extend(lcv_terms_nodes)
+            #nodes_to_visit.extend(lcv_terms_nodes)
+        #return lcv_terms   
+
+class CounterNode(StructuredNode):
+    counter = IntegerProperty(default=0)
+    updated_at = DateTimeProperty(default=lambda: datetime.now())
+
+    @classmethod
+    def get(cls):
+        counter_node = cls.nodes.first_or_none()
+        if counter_node is None:
+            return cls.create_node()
+        return counter_node
+
+    @classmethod
+    def create_node(cls):
+        counter = cls()
+        counter.save()
+        return counter
+    
+    @classmethod
+    def increment(cls):
+        counter = cls.get()
+        counter.counter += 1
+        counter.updated_at = datetime.now()
+        counter.save()
+        return counter
+
+
+class Provider(models.Model):
+    uid = models.CharField(default=lambda: uid_generator.generate_uid(), max_length=36, editable=False, unique=True)
+    #uid = models.CharField(default=lambda: generate_uid(str(datetime.now())), max_length=36, editable=False, unique=True)
+    #uid = models.UUIDField(default=uuid.uuid4, editable=False, unique=True)
+    name = models.CharField(max_length=255)
+
+class LCVTerm(models.Model):
+    uid = models.CharField(default=lambda: uid_generator.generate_uid(), max_length=36, editable=False, unique=True)
+    #uid = models.CharField(default=lambda: generate_uid(str(datetime.now())), max_length=36, editable=False, unique=True)
+    #uid = models.UUIDField(default=uuid.uuid4, editable=False, unique=True)
+    term = models.CharField(max_length=255)
+    ld_lcv_structure = models.CharField(max_length=255)  # Adjust as needed
+
+#class LanguageSet(models.Model):  # Assuming this is the model for xBOM
+ #   uid = models.CharField(default=lambda: uid_generator.generate_uid(), max_length=36, editable=False, unique=True)
+  #  name = models.CharField(max_length=255)
+  #  terms = models.ManyToManyField(LCVTerm, related_name='language_sets')
+
+# LanguageSet now a DjangoNode
+class LanguageSet(StructuredNode):
+    uid = StringProperty(default=lambda: uid_generator.generate_uid(), unique_index=True)
+    name = StringProperty(required=True)
+    terms = RelationshipTo(LCVTerm, 'HAS_TERM')
```

Comment:
Where does this fit into the creation of nodes? When would we want to create a new LanguageSet? Is this to store the LCVID?

Reply:
Language Sets starts to get into the xBOM Territory which is future work; This effort we are just focused on UID and CCV. We could make just a basic node for that future work that houses language set and links to all the term(LCV_IDs)